.github/actions/install-deps/action.yml

```diff
@@ -0,0 +1,98 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+name: Install dependencies
+description: Install system dependencies needed for OpenTitan
+
+inputs:
+  verilator-version:
+    description: Verilator version to install
+    required: true
+    default: '4.210'
+  verilator-path:
+    description: Path at which to install Veriltator
+    required: true
+    default: /tmp/verilator
+  verible-version:
+    description: Verible version to install
+    required: true
+    default: 'v0.0-3622-g07b310a3'
+  verible-path:
+    description: Path at which to install Verible
+    required: true
+    default: /tmp/verible
+  apt-archive-path:
+    description: Path at which to download APT package archives
+    required: true
+    default: /tmp/apt-archives
+
+runs:
+  using: composite
+  steps:
+    - name: Enable caching for APT packages
+      uses: actions/cache@v4
+      with:
+        # We cache the directory where APT downloads `.deb` archives for packages.
+        path: ${{ inputs.apt-archive-path }}
+        key: ${{ runner.os }}-${{ hashFiles('apt-requirements.txt') }}
+      id: cache-apt
+
+    - name: Install system dependencies
+      run: |
+        mkdir -p "${{ inputs.apt-archive-path }}"
+        # Download archives separately, otherwise APT deletes them once installed.
+        grep '^[^#]' apt-requirements.txt | \
+          xargs sudo apt -o Dir::Cache::archives=${{ inputs.apt-archive-path }} install --download-only
+        grep '^[^#]' apt-requirements.txt | \
+          xargs sudo apt -o Dir::Cache::archives=${{ inputs.apt-archive-path }} install -y
+        # Allow the cache-saving step to read the directory contents.
+        sudo chmod -R +rw ${{ inputs.apt-archive-path }}
+      shell: bash
+
+    - uses: actions/setup-python@v5
+      with: # TODO: use `python-version-file` when we have `pyproject.toml`.
+        python-version: 3.9
+        cache-dependency-path: python-requirements.txt
+        cache: pip
+
+    - name: Install Python dependencies
+      run: python3 -m pip install -r python-requirements.txt --require-hashes
```

Comment:
Could change to uv

Reply:
Just tried this, but `uv` will only install into a `venv` and not the system. I don't want to have to `source .venv/bin/activate` and the start of every workflow step, and the speed increase wasn't very significant anyway.